lib/modules/manager/gradle/artifacts.ts

```diff
@@ -148,13 +154,26 @@ export async function updateArtifacts({
       .map(quote)
       .join(' ')}`;
 
+    if (verificationMetadataFile) {
+      const hashTypes: string[] = [];
+      const verificationMetadata = await readLocalFile(
+        verificationMetadataFile
+      );
+      ['md5', 'sha1', 'sha256', 'sha512', 'pgp'].forEach((hashType) => {
```

Comment:
If someone sets up their verification metadata using unsafe checksums, I would expect Renovate to update them. But if you view this differently I will follow what you say, since I have no need for the outdated checksums myself.

Simply not updating them will lead to verification errors after dependency updates, though. How would you warn users about what is happening?

Reply:
just use sha256 is there is a weak hash algorithm was used.

Reply:
If you update existing entries that have both `sha1` and `sha256` with `./gradlew --console=plain -q --write-verification-metadata sha256 help` would that ignore the existing `sha1` or delete it?

Reply:
It won't delete anything, it won't even delete the sha256 if it changes for some reason. It will leave the (potentially wrong) sha1 and add the new sha256 value as `<also-trust>` element.

In my opinion this is fine, since this is just how gradle does this. The hash for an existing artifact shouldn't change and if it does, it should be checked manually anyway.

Reply:
Agreed. From a user perspective, it's counter-intuitive that renovate ignores `md5` or `sha1` digests but apart from that, the whole verification metadata functionality makes no sense if used with hash functions that are not collision resistant.